lighthouse-core/gather/driver.js

```diff
@@ -1477,6 +1476,31 @@ class Driver {
     }
   }
 
+  /**
+   * @param {string} url
+   * @return {Promise<string[]>}
+   */
+  async getImportantLocationsNotCleared(url) {
+    const usageData = await this.sendCommand('Storage.getUsageAndQuota', {
+      origin: url,
+    });
+    const locations = usageData.usageBreakdown.filter(usage => usage.usage)
+      .map(usage => {
+        switch (usage.storageType) {
+          case 'local_storage':
+            return 'Local Storage';
+          case 'indexeddb':
+            return 'IndexedDB';
+          case 'websql':
+            return 'Web SQL';
+          default:
+            return '';
+        }
+      })
+      .filter(resourceString => resourceString);
```

Comment:
something like
```js
const storageTypeNames = {
  local_storage: 'Local Storage',
  indexeddb: 'IndexedDB',
  websql: 'Web SQL',
};
const locations = usageData.usageBreakdown
  .filter(usage => usage.usage)
  .map(usage => storageTypeNames[usage.storageType] || '')
  .filter(Boolean);
```

might be a little clearer in what the code is up to? I assume there will be a tsc issue to be resolved on the `storageTypeNames[usage.storageType]` part, though.